main.tf

```diff
@@ -41,25 +41,27 @@ resource "random_string" "deployment_id" {
 }
 
 resource "azurerm_resource_group" "aap" {
-  name     = var.resource_group
+  name = var.resource_group
   location = var.location
-  tags     = local.persistent_tags
+  tags = local.persistent_tags
 }
 
 resource "azurerm_virtual_network" "aap" {
-  name                = "${var.deployment_id}-aap-vn"
-  location            = azurerm_resource_group.aap.location
+  depends_on = [ azurerm_resource_group.aap ]
+  name = "${var.deployment_id}-aap-vn"
+  location = azurerm_resource_group.aap.location
   resource_group_name = azurerm_resource_group.aap.name
-  address_space       = [var.infrastructure_vpc_cidr]
-  tags                = local.persistent_tags
+  address_space = [var.infrastructure_vpc_cidr]
+  tags = local.persistent_tags
 }
 
 resource "azurerm_subnet" "aap" {
-  name                 = "${var.deployment_id}-aap-sn"
+  depends_on = [ azurerm_virtual_network.aap ]
+  name = "${var.deployment_id}-aap-sn"
```

Comment:
Maybe we should change `-sn` to `-subnet` to match the new one

Reply:
I will change it here. But this will go in the network module once we finish network story.